### PR TITLE
Add Linux generic support

### PR DESCRIPTION
**Pull request by dubek**
The new Linux generic support is actually the original code from `lstools_ubuntu`.

Now `MbedLsToolsUbuntu` inherits from `MbedLsToolsLinuxGeneric` (to allow Ubuntu-specific changes in the future). Supposedly since they are identical we can remove the Ubuntu-specific class which adds no functionality right now, but I understand the mbed dev team wants to keep Ubuntu separate.

I tested this on RHEL 6.7 with Freescale FRDM-K64F.

Maybe solves #25.

### CODE REVIEW

**Comment by PrzemekWirkus** [comment]
Thank you for your contribution. I will have a look at this shortly!
Did sing [contributor agreement](https://developer.mbed.org/contributor_agreement/) ?

**Comment by dubek** [comment]
hmmm, let me consult my lawyer :-)

**Comment by PrzemekWirkus** [comment]
;)

PS: I can merge it after TechCon (mid of November) most probably because for now we have a feature freeze on this software.

File: mbed_lstools/lstools_linux_generic.py [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+        """ Find mount points for MBED devices using mount command output
+
+        @return Returns None if mount point not found. Else returns device mount path
+
+        @param dev_name Device name (e.g 'sda')
+        @param mount_list List of all mounted devices (strings from Linux mount shell command)
+
+        @details
+        """
+        mount_media_pattern = "^/[a-zA-Z0-9/]*/" + dev_name  + " on (/[a-zA-Z0-9/]*) "
```
Comment (jupe) [inline mbed_lstools/lstools_linux_generic.py]: there should be also `_` in end of detection pattern, like:
`"^/[a-zA-Z0-9/]*/" + dev_name  + " on (/[a-zA-Z0-9_/]*) "`.
This because looks that when multiple devices are connected mount point will be 
* `/media/MBED_` for second device
* `/media/MBED__` for third device
and so on...
* `/media/MBED______`  ... 
.

File: mbed_lstools/lstools_ubuntu.py [hunk 2]
```diff
@@ -15,297 +15,14 @@
...
 
-import re
-import subprocess
+from lstools_linux_generic import MbedLsToolsLinuxGeneric
 
-from lstools_base import MbedLsToolsBase
 
-
-class MbedLsToolsUbuntu(MbedLsToolsBase):
+class MbedLsToolsUbuntu(MbedLsToolsLinuxGeneric):
```
Comment (jupe) [inline mbed_lstools/lstools_ubuntu.py]: is this needed at all, because MbedLsToolUbuntu is now direct inheritance without any special functionality ?

**Comment by dubek** [comment]
@PrzemekWirkus I signed the contributor agreement.

File: mbed_lstools/lstools_ubuntu.py (hunk 2 above)
Comment (dubek) [inline mbed_lstools/lstools_ubuntu.py]: The only reason is this comment https://github.com/ARMmbed/mbed-ls/issues/25#issuecomment-151058043 - where @PrzemekWirkus stated that Ubuntu is the main testing platform.

File: mbed_lstools/lstools_linux_generic.py (hunk 1 above)
Comment (dubek) [inline mbed_lstools/lstools_linux_generic.py]: You're right, but that's another issue. Anyway I need to rebase this branch (carefully) because there have been other changes to the code (that I extracted) in the meantime.

**Comment by dubek** [comment]
I pushed this again rebased on top of current master branch.

File: mbed_lstools/lstools_linux_generic.py (hunk 1 above)
Comment (dubek) [inline mbed_lstools/lstools_linux_generic.py]: @jupe - I think you should open a separate PR about your pattern change for multiple devices.

File: mbed_lstools/lstools_ubuntu.py (hunk 2 above)
Comment (jupe) [inline mbed_lstools/lstools_ubuntu.py]: I see

File: mbed_lstools/lstools_linux_generic.py (hunk 1 above)
Comment (jupe) [inline mbed_lstools/lstools_linux_generic.py]: yep, it was in original code too, so this need to be done in separate PR.

Comment (jupe) [inline mbed_lstools/lstools_linux_generic.py]: https://github.com/ARMmbed/mbed-ls/pull/33
